### ISSUE TITLE
Demo crashes if browser has restrictive storage settings

### ISSUE THREAD

**Issue by thanatos** [firehol/netdata#1714]
If you visit [the demo](https://london.my-netdata.io/default.html), it doesn't display anything

![it's blank](http://i.imgur.com/u4aH6I0.png)

This is b/c the application crashes in JS code, failing to access `localStorage`:

```
Uncaught DOMException: Failed to read the 'localStorage' property from 'Window': Access is denied for this document.
    at Object.NETDATA.localStorageGet (https://london.my-netdata.io/dashboard.js?v20170127-1:374:27)
    at Object.NETDATA.localStorageGetRecursive (https://london.my-netdata.io/dashboard.js?v20170127-1:442:30)
    at https://london.my-netdata.io/dashboard.js?v20170127-1:476:13
    at https://london.my-netdata.io/dashboard.js?v20170127-1:6639:3
```

I'm a big believer in not being tracked; to that end, I whitelist cookies. I don't believe a demo should require storage, and an application should fall back gracefully. I don't expect anything to be saved, of course, if such a feature exists. At the very least, if cookies are required (or being forced upon the user), an error message to that effect is nicer than crashing. (Though obviously, I prefer the graceful fallback.)

The code around this point seems to do a decent job of checking if the browser is localStorage capable, and falling back if not; it just seems to forget the case of access being denied. (And note that denial happens automatically with no visual prompt in common browsers, such as Chrome.)

**Comment by ktsaou** [comment]
hi, the demo requires localstorage to store dashboard settings (i.e. theme, various other settings, chart size - if you change them of course), etc.

I didn't know such an error could exist. I am fixing it...

**Comment by ktsaou** [comment]
fixed in PR #1713
